### PR TITLE
add normalized scores to racket server

### PR DESCRIPTION
**Pull request by gprice1129**

### CODE REVIEW

File: trapi.rkt [hunk 1]
```diff
@@ -330,11 +332,30 @@
...
+
+  (define (merge-scores r s)
+    (define current-scores (summary-fragment-scores s))
+    (define new-score (car (hash->list (summary-fragment-scores r))))
+    (define new-score-key (car new-score))
+    (define key-scores (jsexpr-object-ref current-scores new-score-key (jsexpr-array)))
+    (jsexpr-object-set
+      current-scores
+      new-score-key
+      (append `(,(cdr new-score)) key-scores)))
```
Comment (gregr) [inline trapi.rkt:352]: `(cons (cdr new-score) key-scores)`

File: trapi.rkt [hunk 2]
```diff
@@ -330,11 +332,30 @@
...
   (string->symbol (number->string (equal-hash-code path))))
 
 (define (merge-summary-attrs r s)
-  (map (lambda (r-attr s-attr) (append r-attr s-attr)) r s))
+  (define (merge-lists r s accessor)
+    (append (accessor r) (accessor s)))
+
+  (define (merge-scores r s)
+    (define current-scores (summary-fragment-scores s))
+    (define new-score (car (hash->list (summary-fragment-scores r))))
```
Comment (gregr) [inline trapi.rkt:346]: Is it intentional that you only take the first?  What are you doing with the rest of them?

File: trapi.rkt [hunk 3]
```diff
@@ -727,12 +763,14 @@
...
                                `(,(string->symbol drug) names)))
-                  (disease (last subgraph)))
+                  (disease (last subgraph))
+                  (scores (jsexpr-object-ref scores (string->symbol drug))))
              (make-immutable-hash `((subject   . ,drug)
                                     (drug_name . ,(if (null? drug-name)
                                                     'null
                                                     (car drug-name)))
                                     (paths     . ,(sort ps path<?))
+                                    (score     . ,(/ (foldl + 0 scores) (length scores)))
```
Comment (gregr) [inline trapi.rkt:773]: Are you guaranteed to have at least one score?

**Review by gregr — APPROVED** [review]
If none of my comments alarm you, then this is fine.

File: trapi.rkt (hunk 2 above)
Comment (gprice1129) [reply trapi.rkt:346]: There can only be one score per result.

**Pull request merged**